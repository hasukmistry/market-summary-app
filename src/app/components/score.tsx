import React, { useEffect } from 'react';
import { loadUser } from "../inc/actions";

type Props = {
    currentUser: {
        id: number,
        points: number,
        ip_addr: string,
        timestamp: string,
    },
    setCurrentUser: any,
    currentScore: number,
    setCurrentScore: any,
    votingEnabled: boolean,
    setVotingEnabled: any
};

export const Score: React.FC<Props> = ({ currentUser, setCurrentUser, currentScore, setCurrentScore, votingEnabled, setVotingEnabled } : Props ) => {
    useEffect(() => {
        let asyncEvents = async() => {
            await loadUser(setCurrentUser, setCurrentScore, setVotingEnabled);
        };

        asyncEvents();
    }, [setCurrentUser, setCurrentScore, setVotingEnabled]);
    return (
        <div style={ scoreStyles }>
            Score <span>{ currentScore }</span>
        </div>
    );
};

const scoreStyles = {
    padding: '5px 14px',
    backgroundColor: '#d9edf7',
    border: '1px solid #bce8f1',
    borderRadius: '15px',
    color: '#337ab7',
    fontWeight: '700'
}
