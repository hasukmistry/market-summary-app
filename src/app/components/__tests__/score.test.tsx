import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Score } from '../score';

const shallowRenderer = createRenderer();

describe('<Score />', () => {
  it('should match snapshot', () => {
    shallowRenderer.render(<Score currentUser={{ id: -1, points: 0, ip_addr: '', timestamp: '' }}
        setCurrentUser={{}}
        currentScore={0}
        setCurrentScore={{}}
        votingEnabled={false} 
        setVotingEnabled={{}}/>);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});