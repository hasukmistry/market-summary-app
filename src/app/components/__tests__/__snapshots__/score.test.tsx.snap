// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Score /> should match snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#d9edf7",
      "border": "1px solid #bce8f1",
      "borderRadius": "15px",
      "color": "#337ab7",
      "fontWeight": "700",
      "padding": "5px 14px",
    }
  }
>
  Score 
  <span>
    0
  </span>
</div>
`;
